Library/Homebrew/os/mac/keg.rb

```diff
@@ -31,4 +33,37 @@ def change_install_name(old, new, file)
     EOS
     raise
   end
+
+  def apply_ad_hoc_signature(file)
+    return if MacOS.version < :big_sur
```

Comment:
Do we want to also limit this to ARM for now?

Reply:
I am waiting for feedback from @mistydemeo, who is checking whether Intel Big Sur codesigns by default or not.
If we want to limit to ARM, I don't know how to do in terms of Ruby calls. Can you suggest something?

Reply:
Xcode 12.2, even on ARM Big Sur, does not codesign for Intel:

```
% clang -arch x86_64 a.c
% file a.out
a.out: Mach-O 64-bit executable x86_64
% codesign -v a.out
a.out: code object is not signed at all
In architecture: x86_64
```

So I suppose you're right, we should limit to Intel. Help is welcome. Looking at `os/mac/mach.rb` it looks like I should be able to simply call `file.arch`? But maybe it's not handled, as the code does not appear to include Arm variants in the list…

Reply:
@fxcoudert Try `Hardware::CPU.arm?`, it’s in `hardware.rb`.